R/PackageDependencyReporter.R

```diff
@@ -30,11 +60,65 @@ PackageDependencyReporter <- R6::R6Class(
     #TODO [patrick.boueri@uptake.com]: Add version information to dependency structure
 
     public = list(
-        set_package = function(packageName, ...) {
-            private$edges <- ExtractDependencyNetwork(packageName, ...)
-            private$nodes <- data.table::data.table(node = unique(c(private$edges[, SOURCE], private$edges[,TARGET])))
-            private$packageName <- packageName
-            private$pkgGraph <- MakeGraphObject(private$edges,private$nodes)
+
+        calculate_metrics = function(...){
+            private$edges <- self$extract_network(...)
+            private$nodes <- data.table::data.table(node = unique(c(private$edges[, SOURCE], private$edges[, TARGET])))
```

Comment:
This function is only for package dependency networks, right?  If so, there's no chance of unconnected nodes, so this line works.  However, with internal functional dependency networks, rogue nodes exist so this method might miss some nodes.

Reply:
This is for package dependencies only, yeah

Given that, can you see a situation where this would break? Definitely something we should address in a unit test